src/common.cu

```diff
@@ -0,0 +1,22 @@
+#include <cuda_runtime.h>
+#include <stdio.h> 
+
+
+cudaError_t cudaMallocCheck(void **dev_ptr, size_t size) {
+    cudaError_t cuda_malloc_return = cudaMalloc(dev_ptr, size);
+    if (cuda_malloc_return == cudaErrorMemoryAllocation) {
+        printf("CUDA malloc failed!\n");
+        exit(EXIT_FAILURE);
+    }
+    return cuda_malloc_return;
+}
+
+
+void *mallocCheck(size_t size) {
+    void *malloc_return = malloc(size);
+    if (malloc_return == NULL) {
```

Comment:
More of a personal preference thing, but you can use `!` instead of checking against `NULL` to do things like:

```c
if (!malloc_return) {
  printf("Malloc failed!\n");
  exit(EXIT_FAILURE);
}
```